cls/pkg/isc/codetidy/CodeTidy.cls

```diff
@@ -129,556 +153,1219 @@ ClassMethod ProcessStream(ByRef instream, lang = "COS", pInfo)
+			set next("Indent") = next("Indent") + 1
+			if (this("Type") = $$$COSJSONBracket) {
+				do $increment(conv("JSONSwitch", "braceCount"))
+				set conv("JSONSwitch") = 1
 			}
-			//&SQL etc....  watch out - also includes <> () ## forms
-			if tType=$$$COSSQL {
-				set cfl=$zconvert(tCodeFragment,"U")
-				do instream.Write(cfl)
-				continue
+			if (this("Type") = $$$JAVASCRIPTDelimiter)
+				&& conv("JSSwitch")
+				&& (this("Frag") = "{") {
+				if $increment(conv("JSSwitch", conv("JSSwitch"), "braceCount"))
 			}
-			if tType=$$$COSJavascript {
-				set cfl=$zconvert(tCodeFragment,"L")
-				if cfl="js" set cfl="javascript"
-				do instream.Write(cfl)
-				continue
+			
+		} elseif CloseChars [ this("Frag") {
+			set i = 1
+			if (this("Type") = $$$JAVASCRIPTDelimiter)
+				&& conv("JSSwitch")
+				&& (this("Frag") = "}") {
+				if $increment(conv("JSSwitch", conv("JSSwitch"), "braceCount"), -1)
+				if conv("JSSwitch", conv("JSSwitch"), "braceCount") = 0 {
+					set i = 2
+					if $increment(conv("JSSwitch"), -1)
+				}
 			}
-			if tType=$$$COSHTML {
-				set cfl=$zconvert(tCodeFragment,"L")
-				do instream.Write(cfl)
-				continue
+			if conv("JSONSwitch") && (this("Type") = $$$COSJSONBracket) {
+				if $increment(conv("JSONSwitch", "braceCount"), -1)
+				if conv("JSONSwitch", "braceCount") = 0 {
+					set conv("JSONSwitch") = 0
+				}
 			}
-			if tType=$$$COSPreProcessorCommand {
-				set cfl=$zconvert(tCodeFragment,"L")
-				do instream.Write(cfl)
-				continue
+			if conv("Line") = ##class(pkg.isc.codetidy.Utils).GetIndentString() {
+				set this("Indent") = this("Indent") - i
+			} else {
+				set next("Indent") = next("Indent") - i
 			}
-			if tType=$$$COSComment {
-				set cfl=tCodeFragment
-				//leave the ;; as there may be some code which still requires it
-				if $extract(tCodeFragment,1)=";",$extract(tCodeFragment,1,2)'=";;" set cfl="//"_$extract(tCodeFragment,2,*)
-				if $extract(tCodeFragment,1,2)="//" set cfl="//"_$extract(tCodeFragment,3,*)
-				if $extract(tCodeFragment,1,2)="##;" set cfl="//"_$extract(tCodeFragment,4,*)
-				do instream.Write(cfl)
-				continue
+		}
+	}
+}
+
+ClassMethod ParseFragmentCOS(ByRef last, ByRef this, ByRef next, ByRef conv)
+{
+	if ##class(pkg.isc.codetidy.Utils).GetIndent() {
+		// Indentation Adjustments
+		if (this("Type") = $$$COSBrace) || (this("Type") = $$$COSJSONBracket) || (this("Type") = $$$COSJSONDelimiter) {
+			do ..ApplyIndentation(.this, .next, .conv, "({[", "]})")
+		}
+	}
+	// Embedding Languages
+	if (this("Type") = $$$COSEmbeddingOpen) {
+		set this("Type") = last("Type")
+		set conv("Embedding") = last("Type")
+	} 
+	if (this("Type") = $$$COSEmbeddingClose) {
+		set this("Type") = conv("Embedding")
+	}
+	
+	// ##class
+	if this("Type") = $$$COSObjectClass {
+		set conv("Frag") = this("Frag")
+		if ##class(pkg.isc.codetidy.Utils).GetUseCapitals() = 0 {
+			set conv("Frag") = $zconvert(conv("Frag"), "l")
+		} else {
+			set conv("Frag") = $zconvert(conv("Frag"), "w")
```

Comment:
Oops - commented on wrong line. I'm not clear on what we're doing with conv("Frag") here. Even in a "capitalize commands" setting it'd always be ##class, not ##Class, and it looks like we're just appending ##class rather than using  conv("Frag")

Reply:
Yes, I am not quite sure why it's that way, but I got that from Trak's code. I am just going to remove that codeblock.